File: MachineLearning/data_preprocessing_template.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Thu Jan 11 14:55:43 2018

@author: jamesponwith
"""

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Data.csv')

# Matrix of features
X = dataset.iloc[:, :-1].values

# Create dependent variable martix
Y = dataset.iloc[:,3].values

# Splitting dataset into the Training set and Testing set
from sklearn.cross_validation import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 0.2, random_state = 0)

"""
# Feature Scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.fit_transform(X_test)

"""

File: MachineLearning/Part 1 - Data Preprocessing/data_preprocessing_template.R
# Data Preprocessing

# Importing data set
dataset = read.csv('Data.csv')
# dataset = dataset[, 2:3]

# Splitting dataset into Training set and Testing set
install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Purchased, SplitRatio = 0.8) # input % towards training
training_set = subset(dataset, split == TRUE)
testing_set = subset(dataset, split == FALSE)

# Feature Scaling
# training_set[, 2:3] = scale(training_set[, 2:3])
# testing_set[, 2:3] = scale(testing_set[, 2:3])


